### ISSUE TITLE
Performance metrics for Deepchem Multitask model

### ISSUE THREAD

**Issue by saurabhclusterone** [deepchem/deepchem#1427]
Hello Everyone,

I was able to run the Deepchem Multitask Estimator code in Multi GPU parallel distributed training mode. I had to make few changes but I have a stand alone example code that I am able to run in distributed mode. The distributed training is good to the extent that I get linear speed up with increase in number of worker nodes.

I will be contributing this code to the repo soon.

The original Deepchem Multitask Model using Estimator code that I am referring to is this one:

https://deepchem.io/docs/notebooks/Estimators.html

So, in the above notebook, there is a "mean_auc" function that is passed as metric in the model.make_estimator function. But, I do not see the mean_auc getting printed when I run the example as is in single node instance.

Also, when I myself tried to print or log accuracy as a metric, what I came across is the need to add following code to model_fn function in the make_estimator function:

logging_hook = tf.train.LoggingTensorHook({"loss" : loss, 
    "accuracy" : accuracy}, every_n_iter=10)

So, my question is, for Multitask model using estimator, does Deepchem have any logging functionality to print mean_auc or to print accuracy metrics?

If there is one please let me know. Currently I do not see Deepchem functionality to print accuracy metrics using Estimator code specifically.

Thanks

**Comment by peastman** [comment]
You're right, it doesn't provide a way to do this.  It might be reasonable to add optional arguments to `TensorGraph.make_estimator()` to let you specify more of the properties for EstimatorSpec.  Possibly even just let the user provide arbitrary keyword arguments, and we pass them on to EstimatorSpec, since there seem to be a lot of them and I wouldn't be surprised if they add more in the future.

**Comment by rbharath** [comment]
Pinging this old issue. Since we're swapping to Keras defaults, are there improvements we need to make or does the default Keras->Estimator conversion cover us?

**Comment by rbharath** [comment]
Doing some digging, it looks like the standard "blessed" path is to just use Keras models for everything. I've talked to a couple people who've used them in production and they seem to be stable. I believe Estimator support is only left for backwards compatibility. I'm going to go ahead and close this repo for that reason.
